ckanext/dataset_subscriptions/actions.py

```diff
@@ -39,8 +42,13 @@ def dms_notification_provider(user_dict, since):
                      if activity['user_id'] != user_dict['id']
                      and 'package' in activity['activity_type']]
     # We want a notification per changed dataset, not a list of all changes
-    deduplicated_activity_list = list({item["object_id"]: item for item in activity_list}.values())
+    deduplicated_activity_list = list({item["object_id"]: item for item in sorted(activity_list, key = lambda item: item['timestamp'])}.values())
```

Comment:
too much happening in this line at once, can you split the logic into multiple lines for better readability?
Maybe:
1. sort by timestamp
2. remove duplicates?

Reply:
done